docs/troubleshoot/app-framework/appserver-error-codes.md

```diff
@@ -1800,291 +1800,348 @@ The following error message codes may appear on the app-server log. Use the foll
+  switch case warning message as the unknown _defaultBehavior_.
 
   **Action:**
 
-  TODO
+  No action is needed, but the warning may be needed to debug
 
 
 
-### ZWED0075W":"Services in plugin=%s war grouping skipped. Plugin missing or already grouped
+### ZWED0075W
+
+  Services in plugin=_plugin[i]_ war grouping skipped. Plugin missing or already grouped
 
   **Reason:**
 
-  TODO
+  If element from _plugins_ array, does not contain a compatible service (data service plugin/war service)
+  then the element is skipped and not added to a new array.
 
   **Action:**
 
-  TODO
+  Review the warning message and check for the _plugins_ array element to fix the error
+
 
 
+### ZWED0076W
 
-### ZWED0076W":"Skipping invalid plugin group=
+  Skipping invalid plugin group=_plugins_
 
   **Reason:**
 
-  TODO
+  If the _plugins_ is not an array and the size is less than zero, then it will log a
+  warning message in the else block.
 
   **Action:**
 
-  TODO
+  Make sure _plugins_ should be an array of size greater than zero.
+
 
 
+### ZWED0077W
 
-### ZWED0077W":"Could not extract war for service=%s, error=
+  Could not extract war for service=_key-value_, error=_errorMessage_
 
   **Reason:**
 
-  TODO
+  If the service with the _key-value_ pair is unable to extract the war file then it throws the _errorMessage_
 
   **Action:**
 
-  TODO
+  Check if your war file is in a valid format or exists.
 
 
 
-### ZWED0078W":"Could not access files to determine status for service=%s, error=
+### ZWED0078W
+
+  Could not access files to determine status for service=_key-value_, error=_errorMessage_
 
   **Reason:**
 
-  TODO
+  If we are unable to get the status of war extracted or not, then it throws _errorMessage_ in catch block.
 
   **Action:**
 
-  TODO
+  Check if your war file is in a valid format or not.
+
 
 
+### ZWED0079W
 
-### ZWED0079W":"Cannot add servlet for service=%s, error=
+  Cannot add servlet for service=_key-value_, error=_errorMessage_
 
   **Reason:**
 
-  TODO
+  If unable to add servlet for service _key-value_, then it logs a warning _errorMessage_.
 
   **Action:**
 
-  TODO
+  No action is needed, but the warning may be needed to debug
+
 
 
+### ZWED0080W
 
-### ZWED0080W":"Cannot add servlet for service=%s
+  Cannot add servlet for service=_key-value_
 
   **Reason:**
 
-  TODO
+  When we are not able to get the _dir_ to add servlet for service _key-value_.
 
   **Action:**
 
-  TODO
+  Check if your _dir_ is exists / in valid format or not.
 
 
 
-### ZWED0081W":"Could not start tomcat, error=
+### ZWED0081W
+
+  Could not start tomcat, error=_errorMessage_
 
   **Reason:**
 
-  TODO
+  If tomcat manager is unable to start tomcat with _DOptionsArray_, then it logs _errorMessage_ in catch block.
 
   **Action:**
 
-  TODO
+  Review the warning message and verify _DOptionsArray_ element values are valid or not.
+
 
 
+### ZWED0082W
 
-### ZWED0082W":"%s stderr=%s
+  _this.getIdString()_ stderr=_data_
```

Comment:
Could you please confirm this with the team? I am not really sure if we need to add the function name here.

Reply:
@sakshibobade21 is right here, it's not the best way to log this message. This method points to something.
```
private getIdString(): string {
    return `Tomcat PID=${this.tomcatProcess.pid}:`;
  }

```
The message should be:

Tomcat PID=_pid_: stderr=_data_